server/src/main/java/org/elasticsearch/index/IndexSettings.java

```diff
@@ -697,7 +697,12 @@ public IndexSettings(final IndexMetadata indexMetadata, final Settings nodeSetti
         this.indexMetadata = indexMetadata;
         numberOfShards = settings.getAsInt(IndexMetadata.SETTING_NUMBER_OF_SHARDS, null);
         mode = isTimeSeriesModeEnabled() ? scopedSettings.get(MODE) : IndexMode.STANDARD;
-        this.timestampBounds = mode.getTimestampBound(scopedSettings);
+        this.timestampBounds = mode.getTimestampBound(indexMetadata);
+        if (mode == IndexMode.TIME_SERIES) {
```

Comment:
I think it's enough to do this if it isn't null, right?

Reply:
Again, I'm just paranoid about every hard check here. I *think* it'll make the code easier to reason about.

Reply:
If someone uses `IndexMode.STANDARD` then we also attempt to get the time bounds. But that is ok?

Reply:
o wait, just doing `timestampBounds != null` is fine here.